src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/dllmain.cpp

```diff
@@ -302,11 +302,12 @@ HRESULT WINAPI RoResolveNamespaceDetour(
     DWORD* subNamespacesCount,
     HSTRING** subNamespaces)
 {
+    HSTRING packageGraphDirectory[1];
+    packageGraphDirectory[0] = Microsoft::WRL::Wrappers::HStringReference(exeFilePath.c_str()).Get();
     HRESULT hr = TrueRoResolveNamespace(name, Microsoft::WRL::Wrappers::HStringReference(exeFilePath.c_str()).Get(),
-        packageGraphDirsCount, packageGraphDirs,
+        1, packageGraphDirectory,
         metaDataFilePathsCount, metaDataFilePaths,
         subNamespacesCount, subNamespaces);
-
```

Comment:
This references a temporary created that's destroyed immediately at the `.Get();`

Instead, use:

```c++
auto pathReference = Microsoft::WRL::Wrappers::HStringReference(exeFilePath.c_str());
HSTRING packageGraphDirectories[] = { pathReference.Get() };
```

And then use `ARRAYSIZE(packageGraphDirectories), packageGraphDirectories` as the parameter pair below.